src/python/grpcio/grpc/experimental/aio/_channel.py

```diff
@@ -13,42 +13,111 @@
 # limitations under the License.
 """Invocation-side implementation of gRPC Asyncio Python."""
 import asyncio
-from typing import Callable, Optional
+from typing import Any, Optional, Sequence, Text, Tuple
 
+import grpc
 from grpc import _common
 from grpc._cython import cygrpc
+from ._call import Call, UnaryUnaryCall, UnaryStreamCall
+from ._typing import (DeserializingFunction, MetadataType, SerializingFunction)
 
-from ._call import Call
 
-SerializingFunction = Callable[[str], bytes]
-DeserializingFunction = Callable[[bytes], str]
+def _timeout_to_deadline(loop: asyncio.AbstractEventLoop,
+                         timeout: Optional[float]) -> Optional[float]:
+    if timeout is None:
+        return None
+    return loop.time() + timeout
 
 
 class UnaryUnaryMultiCallable:
-    """Afford invoking a unary-unary RPC from client-side in an asynchronous way."""
+    """Factory an asynchronous unary-unary RPC stub call from client-side."""
 
     def __init__(self, channel: cygrpc.AioChannel, method: bytes,
                  request_serializer: SerializingFunction,
                  response_deserializer: DeserializingFunction) -> None:
+        self._loop = asyncio.get_event_loop()
         self._channel = channel
         self._method = method
         self._request_serializer = request_serializer
         self._response_deserializer = response_deserializer
-        self._loop = asyncio.get_event_loop()
 
-    def _timeout_to_deadline(self, timeout: int) -> Optional[int]:
-        if timeout is None:
-            return None
-        return self._loop.time() + timeout
+    def __call__(self,
+                 request: Any,
+                 *,
+                 timeout: Optional[float] = None,
+                 metadata: Optional[MetadataType] = None,
+                 credentials: Optional[grpc.CallCredentials] = None,
+                 wait_for_ready: Optional[bool] = None,
+                 compression: Optional[grpc.Compression] = None) -> Call:
```

Comment:
Should we say explicitly that we return a `UnaryUnaryCall` object?

Reply:
Good catch!
Changed to the abstract class of `UnaryUnaryCall`. Also changed for `UnaryStreamCall`.